### DISCUSSION TITLE
Any plans to certify Home Assistant ZHA integration as a CSA certified Zigbee product?

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#1294] [ZHA development]
Wondering if [Nabu Casa](https://www.nabucasa.com/about/) and Home Assistant developers have a plan for getting Home Assistant's [ZHA integration](https://www.home-assistant.io/integrations/zha) solution (and its zigpy dependecies) ready for going through the Zigbee 3.0 certification process as a Zigbee gateway "software component" product?

https://csa-iot.org/certification/transfer-program/

https://csa-iot.org/certification/why-certify/

Zigbee "software component" products are hubs/bridges/controllers and other Zigbee gateway solutions that certificate compatibinitly and interoperatibity with Zigbee 3.0 specifications, so believe it should at least be possible to certify Home Assistant's [ZHA integration](https://www.home-assistant.io/integrations/zha) when running on an applicance like the [Home Assistant Yellow](https://www.home-assistant.io/yellow/) (and perhaps also the [Home Assistant Green](https://www.home-assistant.io/green/) applicance in combination with [Home Assistant SkyConnect](https://www.home-assistant.io/skyconnect/) as well?). Existing of a such products are for examples the SmartThings Hub V2/V3 by Samsung:

https://csa-iot.org/csa-iot_products/?p_keywords=&p_type%5B%5D=1053&p_certificate

https://csa-iot.org/csa_product/smartthings-hub-software-on-hub-v2-v3-2/

Have recently seen that Nabu Casa / Home Assistant developers are already working on their plan for certifying its Z-Wave solution, so a related tip is that the zwave-js organization on GitHub project has a project (kanban-board) with roadmap and backlog tracking tasks needed to be implemented or changed before they can go through the Z-Wave Certification Program:

https://github.com/orgs/zwave-js/projects/1/views/1

https://github.com/zwave-js/certification-backlog/

The reason for asking about this now is that read Nabu Casa has joined the CSA (Connectivity Standards Alliance) as a member, and Home Assistant's founder Paulus Schoutsen has previously also made it clear that Nabu Casa has a goal of certifying Home Assistant's Matter integration as a 100% compatible Matter controller.

https://www.home-assistant.io/blog/2023/12/04/nabu-casa-at-the-matter-member-meeting/

"_**CSA membership ensures that we have access to official technical documentation and support to build Matter into Home Assistant. It also gives us a voice inside the CSA, which we use to advocate for the interests of Home Assistant users and our Open Home vision.**_"

https://www.home-assistant.io/blog/2023/02/08/state-of-matter-and-thread/

Another tip is that Zigbee SoC manufactuers have have documentation and tools to help test implementations based on their hardware, some of which might still be useful to testing vendor and hardware-independent Zigbee gateway solutions (probably at least the Zigbee Coordinator related parts):

https://www.silabs.com/documents/public/application-notes/an1118-certifying-zigbee-30-devices.pdf

https://software-dl.ti.com/simplelink/esd/simplelink_cc13x2_26x2_sdk/3.30.00.03/exports/docs/zigbee/html/zigbee/product_certification.html

PS: CSA also have Authorized Test Service Providers (third-party companies that act as independent testing houses):

https://csa-iot.org/certification/testing-providers/

**Comment by puddly** [comment]
There are no plans to do this at the moment.

**Reply by MattWestb** [reply]
I think its not possible then some of the radio libs cant doing true Zigbe R22 and with lesser security is not possible and only doing some of the radios is useless form user perspective and also code point.
Also like TI firmware is parched so cant being certified and DE i not doing it with there firmware so only EZSP is possible what i can see.

Edit: All firmware must being certificated and the host system code and must being at the right version or its not valid so looks being nearly impossible to do then users can doing other strange things and if some other devs is doing commitment in HA that like dependencies is being one braking cert changes and the cert must being redone (can working if ZHA is being platinum integration but its not for the moment).

**Reply by Hedda** [reply]
> There are no plans to do this at the moment.

OK, I understand but maybe not close this discussion? This was not posted under issues but under [GitHub discussions](https://docs.github.com/en/discussions) section which is meant to be a forum as as such the original post was only the start of a informal open-ended conversation, a continuous dialoge with follow-up questions like what do you think is the minimum viable product (MVP) features and functions needed to get certified? And how many things are still missing in ZHA/zigpy if any?

**Reply by Hedda** [reply]
> > I think its not possible then some of the radio libs cant doing true Zigbe R22 and with lesser security is not possible and only doing some of the radios is useless form user perspective and also code point.

While agree that it is progbably not possible to get zigpy radio libraries certified in general, I think that it should be possibe to certify a specific all-in-one implementation such as the Home Assistant Yellow which has all the hardware and software compnents needed within the same package ->https://www.home-assistant.io/yellow/ ...or do you not believe so?

**Reply by MattWestb** [reply]
**IF**  doing then one EZSP and / or RCP firmware (easy doing for all NC MG21 devices) and one locked ZHA version but must also sealing the firmware and the ZHA so its cant being manipulated.
Its very much work and its also expensive for doing the external cet tests and applying for it and much work and gaining very little and i think NC is not having the resource for doing it (we only having one **Puddly**).

**Reply by Hedda** [reply]
Maybe now that Silicon Labs has partnered with Nabu Casa they as a company could try getting Silabs to sponsor Nabu Casa's employees or consultants time for a Zigbee 3.0 certification project and time at Authorized Test Service Providers to help Validate Certification Eligibility?

https://www.home-assistant.io/blog/2023/12/12/silicon-labs-official-partnership-nabu-casa/

@puddly can you please re-open this discussion thread regardless? Again, it is just meant to be an open-ended discussion.
